src/solver/infeasible-problem-analysis/problem.cpp

```diff
@@ -11,80 +11,179 @@ namespace Antares
+        return;
+    }
+
+    hasDetectedInfeasibilityCause_ = true;
 }
 
-void InfeasibleProblemAnalysis::addSlackVariables()
+void SlackVariablesAnalysis::addSlackVariables()
 {
     /* Optimization:
        We assess that less than 1 every 3 constraint will match
        the regex. If more, push_back may force the copy of memory blocks.
        This should not happen in most cases.
     */
     const unsigned int selectedConstraintsInverseRatio = 3;
-    mSlackVariables.reserve(mSolver->NumConstraints() / selectedConstraintsInverseRatio);
+    slackVariables_.reserve(problem_->NumConstraints() / selectedConstraintsInverseRatio);
     std::regex rgx(constraint_name_pattern);
     const double infinity = MPSolver::infinity();
-    for (MPConstraint* constraint : mSolver->constraints())
+    for (MPConstraint* constraint : problem_->constraints())
     {
         if (std::regex_search(constraint->name(), rgx))
         {
             if (constraint->lb() != -infinity)
             {
                 const MPVariable* slack
-                  = mSolver->MakeNumVar(0, infinity, constraint->name() + "::low");
+                  = problem_->MakeNumVar(0, infinity, constraint->name() + "::low");
                 constraint->SetCoefficient(slack, 1.);
-                mSlackVariables.push_back(slack);
+                slackVariables_.push_back(slack);
             }
 
             if (constraint->ub() != infinity)
             {
                 const MPVariable* slack
-                  = mSolver->MakeNumVar(0, infinity, constraint->name() + "::up");
+                  = problem_->MakeNumVar(0, infinity, constraint->name() + "::up");
                 constraint->SetCoefficient(slack, -1.);
-                mSlackVariables.push_back(slack);
+                slackVariables_.push_back(slack);
             }
         }
     }
 }
 
-void InfeasibleProblemAnalysis::buildObjective() const
+void SlackVariablesAnalysis::buildObjective() const
 {
-    MPObjective* objective = mSolver->MutableObjective();
+    MPObjective* objective = problem_->MutableObjective();
     // Reset objective function
     objective->Clear();
     // Only slack variables have a non-zero cost
-    for (const MPVariable* slack : mSlackVariables)
+    for (const MPVariable* slack : slackVariables_)
     {
         objective->SetCoefficient(slack, 1.);
     }
     objective->SetMinimization();
 }
 
-MPSolver::ResultStatus InfeasibleProblemAnalysis::Solve() const
+MPSolver::ResultStatus SlackVariablesAnalysis::Solve() const
 {
-    return mSolver->Solve();
+    return problem_->Solve();
 }
 
-InfeasibleProblemReport InfeasibleProblemAnalysis::produceReport()
+void SlackVariablesAnalysis::printReport()
 {
-    addSlackVariables();
-    if (mSlackVariables.empty())
+    InfeasibleProblemReport report(slackVariables_);
+    report.prettyPrint();
+}
+
+// ============================
+// Variables bounds analysis
+// ============================
```

Comment:
I strongly suggest splitting this file, and renaming the new files accordingly (constraint slack analysis, variable bound check)